### ISSUE TITLE
Compile: error (undeclared variable) in UCX osc component

### ISSUE THREAD

**Issue by tonycurtis** [open-mpi/ompi#7874]
Thank you for taking the time to submit an issue!

## Background information

Error in UCX component during compilation

### What version of Open MPI are you using? (e.g., v3.0.5, v4.0.2, git branch name and hash, etc.)

git master @ 634f67b216fdcce841867f15ecd75538e65702e8

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)

git clone

### If you are building/installing from a git clone, please copy-n-paste the output from `git submodule status`.

```
 4a43c39c89037f52b4e25927e58caf08f3707c33 opal/mca/hwloc/hwloc2/hwloc (hwloc-2.1.0rc2-53-g4a43c39c)
 8283e81d1c0fd078e0f7fa85a383b633c328254b opal/mca/pmix/pmix4x/openpmix (v1.1.3-2431-g8283e81d)
 66c73f74cc4afd4ead5454771d98b5f199b7fe0e prrte (dev-30650-g66c73f74cc)
```

### Please describe the system on which you are running

* Operating system/version: CentOS 7.5.1804
* Computer hardware:  x86
* Network type:  IBV

-----------------------------

## Details of the problem

```
make[2]: Entering directory `/gpfs/projects/ChapmanGroup/src/openmpi/build/ompi/mca/osc/ucx'
depbase=`echo osc_ucx_comm.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/sh ../../../../libtool  --tag=CC   --mode=compile gcc -std=gnu99 -std=gnu99 -DHAVE_CONFIG_H -I. -I../../../../../openmpi-git/ompi/mca/osc/ucx -I../../../../opal/include -I../../../../ompi/include -I../../../../oshmem/include -I../../../../opal/mca/hwloc/hwloc2/hwloc/include/private/autogen -I../../../../opal/mca/hwloc/hwloc2/hwloc/include/hwloc/autogen -I../../../../ompi/mpiext/cuda/c  -I/gpfs/projects/ChapmanGroup/opt/ucx/1.7.0/include -iquote../../../../../openmpi-git -iquote../../../.. -iquote../../../../../openmpi-git/opal/include -iquote../../../../../openmpi-git/ompi/include -iquote../../../../../openmpi-git/oshmem/include -I/gpfs/projects/ChapmanGroup/src/openmpi/build/opal/mca/pmix/pmix4x/openpmix/include -I/gpfs/projects/ChapmanGroup/src/openmpi/openmpi-git/opal/mca/pmix/pmix4x/openpmix/include -I/gpfs/projects/ChapmanGroup/src/openmpi/build/opal/mca/event/libevent2022/libevent/include -I/gpfs/projects/ChapmanGroup/src/openmpi/openmpi-git/opal/mca/event/libevent2022/libevent -I/gpfs/projects/ChapmanGroup/src/openmpi/openmpi-git/opal/mca/event/libevent2022/libevent/include -I/gpfs/projects/ChapmanGroup/src/openmpi/build/opal/mca/hwloc/hwloc2/hwloc/include -I/gpfs/projects/ChapmanGroup/src/openmpi/openmpi-git/opal/mca/hwloc/hwloc2/hwloc/include  -I/usr/local/include -I/usr/local/include  -O3 -DNDEBUG -I/gpfs/projects/ChapmanGroup/opt/ucx/1.7.0/include -finline-functions -fno-strict-aliasing -mcx16 -MT osc_ucx_comm.lo -MD -MP -MF $depbase.Tpo -c -o osc_ucx_comm.lo ../../../../../openmpi-git/ompi/mca/osc/ucx/osc_ucx_comm.c &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  gcc -std=gnu99 -std=gnu99 -DHAVE_CONFIG_H -I. -I../../../../../openmpi-git/ompi/mca/osc/ucx -I../../../../opal/include -I../../../../ompi/include -I../../../../oshmem/include -I../../../../opal/mca/hwloc/hwloc2/hwloc/include/private/autogen -I../../../../opal/mca/hwloc/hwloc2/hwloc/include/hwloc/autogen -I../../../../ompi/mpiext/cuda/c -I/gpfs/projects/ChapmanGroup/opt/ucx/1.7.0/include -iquote../../../../../openmpi-git -iquote../../../.. -iquote../../../../../openmpi-git/opal/include -iquote../../../../../openmpi-git/ompi/include -iquote../../../../../openmpi-git/oshmem/include -I/gpfs/projects/ChapmanGroup/src/openmpi/build/opal/mca/pmix/pmix4x/openpmix/include -I/gpfs/projects/ChapmanGroup/src/openmpi/openmpi-git/opal/mca/pmix/pmix4x/openpmix/include -I/gpfs/projects/ChapmanGroup/src/openmpi/build/opal/mca/event/libevent2022/libevent/include -I/gpfs/projects/ChapmanGroup/src/openmpi/openmpi-git/opal/mca/event/libevent2022/libevent -I/gpfs/projects/ChapmanGroup/src/openmpi/openmpi-git/opal/mca/event/libevent2022/libevent/include -I/gpfs/projects/ChapmanGroup/src/openmpi/build/opal/mca/hwloc/hwloc2/hwloc/include -I/gpfs/projects/ChapmanGroup/src/openmpi/openmpi-git/opal/mca/hwloc/hwloc2/hwloc/include -I/usr/local/include -I/usr/local/include -O3 -DNDEBUG -I/gpfs/projects/ChapmanGroup/opt/ucx/1.7.0/include -finline-functions -fno-strict-aliasing -mcx16 -MT osc_ucx_comm.lo -MD -MP -MF .deps/osc_ucx_comm.Tpo -c ../../../../../openmpi-git/ompi/mca/osc/ucx/osc_ucx_comm.c  -fPIC -DPIC -o .libs/osc_ucx_comm.o
../../../../../openmpi-git/ompi/mca/osc/ucx/osc_ucx_comm.c: In function 'do_atomic_compare_and_swap':
../../../../../openmpi-git/ompi/mca/osc/ucx/osc_ucx_comm.c:756:18: error: 'temp_addr' undeclared (first use in this function)
             free(temp_addr);
                  ^
../../../../../openmpi-git/ompi/mca/osc/ucx/osc_ucx_comm.c:756:18: note: each undeclared identifier is reported only once for each function it appears in
make[2]: *** [osc_ucx_comm.lo] Error 1
make[2]: Leaving directory `/gpfs/projects/ChapmanGroup/src/openmpi/build/ompi/mca/osc/ucx'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/gpfs/projects/ChapmanGroup/src/openmpi/build/ompi'
make: *** [all-recursive] Error 1
```

**Comment by devreal** [comment]
Oops, that comes from my PR, sorry for that. Will post a fix soon

**Comment by devreal** [comment]
Strange, that `free` was not there in my PR: https://github.com/open-mpi/ompi/pull/6980/files#diff-77ea785e333e9e6e17e99a4460cb88c1R742 Not sure where that comes from but working on a fix.

**Comment by devreal** [comment]
There was a merge conflict between https://github.com/open-mpi/ompi/pull/6980 and https://github.com/open-mpi/ompi/pull/7843 that inserted a stray call to `free`. I wonder why Github didn't complain (I saw it complain in other projects, requiring a rebase before the merge). The fix is in #7875, thanks for catching that @tonycurtis
